src/components/TransactionSentDialog/index.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Typography from 'material-ui/Typography';
+import Button from 'material-ui/Button';
+import Dialog, {
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from 'material-ui/Dialog';
+import { withStyles } from 'material-ui/styles';
+import { injectIntl, intlShape, defineMessages } from 'react-intl';
+
+import styles from './styles';
+
+const messages = defineMessages({
+  successMsg: {
+    id: 'transactionSentDialog.successMsg',
+    defaultMessage: 'Success',
+  },
+  failureMsg: {
+    id: 'transactionSentDialog.failureMsg',
+    defaultMessage: 'Oops, something went wrong',
+  },
+  waitingMsg: {
+    id: 'transactionSentDialog.waitingMsg',
+    defaultMessage: 'Transaction sent. Waiting for confirmations.',
+  },
+  transactionId: {
+    id: 'str.transactionId',
+    defaultMessage: 'Transaction ID',
+  },
+});
+
+class TransactionSentDialog extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.getSuccessText = this.getSuccessText.bind(this);
+    this.getErrorText = this.getErrorText.bind(this);
+  }
+
+  render() {
+    const { classes, txReturn } = this.props;
+
+    let text;
+    if (txReturn) {
+      text = txReturn && txReturn.txid ? this.getSuccessText() : this.getErrorText();
```

Comment:
can we have better names for this block of data?

Reply:
renamed to `textContent`